src/lumigo_opentelemetry/instrumentations/redis/__init__.py

```diff
@@ -0,0 +1,37 @@
+import json
+from typing import List, Any, Dict
+
+from opentelemetry.trace.span import Span
+from lumigo_opentelemetry.instrumentations import AbstractInstrumentor
+from lumigo_opentelemetry.instrumentations.instrumentation_utils import (
+    add_body_attribute,
+)
+
+
+class RedisInstrumentor(AbstractInstrumentor):
+    def __init__(self) -> None:
+        super().__init__("redis")
+
+    def check_if_applicable(self) -> None:
+        import redis  # noqa
+
+    def install_instrumentation(self) -> None:
+        from opentelemetry.instrumentation.redis import RedisInstrumentor
+        from redis.connection import Connection
+
+        def request_hook(
```

Comment:
more out of curiosity than anything, how do these hooks handle pipelining? does each pipeline request get treated as one batch request, or do we get a separate span for each individual request?

Reply:
span per executing a command, because they wrap the function `ClusterPipeline.execute`, so it depends on how you build your app :)